stripe/src/main/java/com/stripe/android/AbstractEphemeralKey.java

```diff
@@ -226,21 +226,32 @@ String getType() {
     protected static <TEphemeralKey extends AbstractEphemeralKey> TEphemeralKey
     fromJson(@Nullable JSONObject jsonObject, Class ephemeralKeyClass) {
         if (jsonObject == null) {
-            return null;
+            throw new IllegalArgumentException("Exception instantiating " +
+                    ephemeralKeyClass.getSimpleName() +
+                    " null JSON");
         }
 
         try {
             return (TEphemeralKey)
                     ephemeralKeyClass.getConstructor(JSONObject.class).newInstance(jsonObject);
         } catch (InstantiationException e) {
-            throw new IllegalArgumentException("Exception instantiating " + ephemeralKeyClass, e);
+            throw new IllegalArgumentException("Exception instantiating " +
+                    ephemeralKeyClass.getSimpleName(), e);
         } catch (IllegalAccessException e) {
-            throw new IllegalArgumentException("Exception instantiating " + ephemeralKeyClass, e);
+            throw new IllegalArgumentException("Exception instantiating " +
+                    ephemeralKeyClass.getSimpleName(), e);
         } catch (InvocationTargetException e) {
-            throw new IllegalArgumentException("Exception instantiating " + ephemeralKeyClass, e);
+            if (e.getTargetException() != null){
+                throw new IllegalArgumentException("Improperly formatted JSON for ephemeral key " +
```

Comment:
I think we can remove the `e.getTargetException().getMessage()` part from the message because we're passing `e.getTargetException()` as an argument.

Reply:
hm, in an ideal world yes, but the interface of `onKeyError` only accepts a String, and that's what we're passing to the dev.

So without explicitely dropping the message in the String, we risk losing it as it is surfaced, if that makes sense